### PR TITLE
OH2-255 Revise and align permissions for CORE, API and UI

### PR DESCRIPTION
**Pull request by mwithi**
See OH2-255.

- [x] remove duplicates
- [x] add missing
- [x] refactor some controllers and endpoints (resources in plurals)
- [x] made all permissions spelled like endpoint
- [x] sort in alphabetical order, then CRUD order
- [x] remove PATCH methods (we are not using as for now, maybe in future)
- [x] move `/discharges` to `/admissions/discharges` endpoint and remove related permissions (`admissions.*` ones will be used)
- [x] update oh.yaml

### CODE REVIEW

**Review by dbmalkovsky — APPROVED** [review]
LGTM (to be honest I didn't review the `oh.yaml` file very much).

File: openapi/oh.yaml [hunk 1]
```diff
@@ -4513,33 +4317,51 @@ paths:
...
+                type: array
+                items:
+                  $ref: '#/components/schemas/MovementDTO'
+      security:
+        - bearerAuth: []
+  /medicalstockmovements/{ref}:
+    get:
+      tags:
+        - Stock Movements
+      operationId: getMovements_1
```
Comment (mwithi) [inline openapi/oh.yaml:4340]: It is not clear to me why we got now a number of `operationsId` of type _1 🤔

**Comment by mwithi** [comment]
@npasquetto after #415 we have to update also https://github.com/informatici/openhospital-ui/pull/538 ?

**Comment by npasquetto** [comment]
> @npasquetto after #415 we have to update also [informatici/openhospital-ui#538](https://github.com/informatici/openhospital-ui/pull/538) ?

Yes, due to the naming refactoring, the SPA API client must be regenerated

File: openapi/oh.yaml (hunk 1 above)
Comment (mwithi) [reply openapi/oh.yaml:4340]: Ok, didn't notice it was there before. Thanks!

**Comment by mwithi** [comment]
> > @npasquetto after #415 we have to update also [informatici/openhospital-ui#538](https://github.com/informatici/openhospital-ui/pull/538) ?
> 
> Yes, due to the naming refactoring, the SPA API client must be regenerated

Ok, let me do that, thanks!

**Comment by mwithi** [comment]
> > > @npasquetto after #415 we have to update also [informatici/openhospital-ui#538](https://github.com/informatici/openhospital-ui/pull/538) ?
> > 
> > 
> > Yes, due to the naming refactoring, the SPA API client must be regenerated
> 
> Ok, let me do that, thanks!

@npasquetto done! please test with https://github.com/informatici/openhospital-ui/pull/538

**Review by npasquetto — APPROVED** [review]
I did some tests and from my point of view it's ok.
I encountered some anomalies but they are not due to the changes in this PR.

**Pull request merged**